### ISSUE TITLE
Relative date off by two days?

### ISSUE THREAD

**Issue by finitenamespace** [allinurl/goaccess#1259]
I've been trying to figure out how to generate a html file for the past 7 days of traffic while ingesting my entire log library, and while not doing what I'm trying to accomplish, I'm wondering why I'm getting relative dates off by ~2 days.

Running:
`zcat -f /var/log/nginx/access.log* | sed -n '/'$(date '+%d\/%b\/%Y' -d '7 days ago')'/,$ p' | goaccess -a -o report.html`
Will generate a report for 01/Oct/2017 — 30/Sep/2018, which is the beginning of my log history to 5 days ago (not 7)

Date output from my console:
```
(date '+%d-%b-%Y' -d '7 days ago')
28-Sep-2018
```

Likewise, running:
`zcat -f /var/log/nginx/access.log* | sed -n '/'$(date '+%d\/%b\/%Y' -d '14 days ago')'/,$ p' | goaccess -a -o report.html`
Generates a report for 01/Oct/2017 — 23/Sep/2018, where I believe it should be from the 21st of September:
```
(date '+%d-%b-%Y' -d '14 days ago')
21-Sep-2018
```

Am I missing something?
Jumping into my /var/log/nginx/access.log files, dates appears like so:
`[30/Sep/2018:06:45:52 +1000]`

And the following log files are present:
```
~# ls /var/log/nginx/
access.log        access.log.34.gz  error.log.1      error.log.35.gz
access.log.1      access.log.35.gz  error.log.10.gz  error.log.36.gz
access.log.10.gz  access.log.36.gz  error.log.11.gz  error.log.37.gz
access.log.11.gz  access.log.37.gz  error.log.12.gz  error.log.38.gz
access.log.12.gz  access.log.38.gz  error.log.13.gz  error.log.39.gz
access.log.13.gz  access.log.39.gz  error.log.14.gz  error.log.3.gz
access.log.14.gz  access.log.3.gz   error.log.15.gz  error.log.40.gz
access.log.15.gz  access.log.40.gz  error.log.16.gz  error.log.41.gz
access.log.16.gz  access.log.41.gz  error.log.17.gz  error.log.42.gz
access.log.17.gz  access.log.42.gz  error.log.18.gz  error.log.43.gz
access.log.18.gz  access.log.43.gz  error.log.19.gz  error.log.44.gz
access.log.19.gz  access.log.44.gz  error.log.20.gz  error.log.45.gz
access.log.20.gz  access.log.45.gz  error.log.21.gz  error.log.46.gz
access.log.21.gz  access.log.46.gz  error.log.22.gz  error.log.47.gz
access.log.22.gz  access.log.47.gz  error.log.23.gz  error.log.48.gz
access.log.23.gz  access.log.48.gz  error.log.24.gz  error.log.49.gz
access.log.24.gz  access.log.49.gz  error.log.25.gz  error.log.4.gz
access.log.25.gz  access.log.4.gz   error.log.26.gz  error.log.50.gz
access.log.26.gz  access.log.50.gz  error.log.27.gz  error.log.51.gz
access.log.27.gz  access.log.51.gz  error.log.28.gz  error.log.52.gz
access.log.28.gz  access.log.52.gz  error.log.29.gz  error.log.5.gz
access.log.29.gz  access.log.5.gz   error.log.2.gz   error.log.6.gz
access.log.2.gz   access.log.6.gz   error.log.30.gz  error.log.7.gz
access.log.30.gz  access.log.7.gz   error.log.31.gz  error.log.8.gz
access.log.31.gz  access.log.8.gz   error.log.32.gz  error.log.9.gz
access.log.32.gz  access.log.9.gz   error.log.33.gz
access.log.33.gz  error.log         error.log.34.gz
```

Aside from that, would I also be able to get a hand figuring out how to format that date string so it grabs x days ago -> the present day (as opposed to x days <- furthest back it can go)?
I'm trying to get a monthly snapshot without having to edit nginx log periods and rollovers.

**Comment by allinurl** [comment]
What is the output of simply running `date`?

**Comment by finitenamespace** [comment]
```
date
Sat Oct  6 00:09:46 AEST 2018
```

Note: It did _just_ tick over to the 6th - was the 5th writing that issue up.

**Comment by allinurl** [comment]
I wonder if this is a sort issue. In the visitors panel (first panel), do you see any Sept. entries within that table?

**Comment by finitenamespace** [comment]
Yep, though there's a gap now that you mention it which I didn't notice.

This is for the <-30 Sept report ("7 days ago"):
![capture](https://user-images.githubusercontent.com/39365213/46541684-040ddb00-c900-11e8-9b61-6b682f82acfc.PNG)

Server definitely wasn't off for more than an hour, and never on consecutive days (certainly not for a 5 day period).

**Comment by allinurl** [comment]
Just to be sure, this is screenshots are pulling data from the same access.log, correct? How many hits (total) has the data set? and I'm assuming there're only 7 days worth of log.

**Comment by finitenamespace** [comment]
> Just to be sure, this is screenshots are pulling data from the same access.log, correct?

Those were the different screenshots from running the "7 days ago"/"14 days ago" from both the 5th of October and the 6th of October. 

This screenshot: 
![capture](https://user-images.githubusercontent.com/39365213/46541684-040ddb00-c900-11e8-9b61-6b682f82acfc.PNG)

Was done using:
`zcat -f /var/log/nginx/access.log* | sed -n '/'$(date '+%d\/%b\/%Y' -d '7 days ago')'/,$ p' | goaccess -a -o report.html`

That pulled 30th of September 2018 to (I'm assuming) my oldest logs of 1st of October 2017. 
But it was run on the _6th_ of October, so I'm assuming the "7 days ago" flag should have pulled the 28/29th of October, not the 30th. Then after you asked for the Hits period is when I also noticed days missing from that list.

Period 01/10/17 to 30/09/18:
<img width="142" alt="capture_period" src="https://user-images.githubusercontent.com/39365213/46771288-9bc75b00-cd3e-11e8-8b78-90439653aa93.PNG">


>How many hits (total) has the data set?

1,733,900
<img width="68" alt="hits_capture" src="https://user-images.githubusercontent.com/39365213/46771290-9e29b500-cd3e-11e8-91db-3bd3c1293bb7.PNG">


> and I'm assuming there're only 7 days worth of log.

No... Although that's actually what I'm ultimately trying to do, but I'm not sure how I'd format that sed date string (that was part 2 of my question after noticing the dates being somewhat off haha). 
I pulled the example straight from the FAQ, but for me it appears to pull ~7 days ago to the beginning of the log files, as opposed to what I want which is, 7 days ago to the most recent day.

And again, just in case it helps, my access logs appear to have the date format of:
`[30/Sep/2018:06:45:52 +1000]`

**Comment by allinurl** [comment]
Can I ask, in the Unique Visitors panel, right in Data column, what's the total there (above the list of dates)?

**Comment by finitenamespace** [comment]
360 Total

![capture](https://user-images.githubusercontent.com/39365213/46897233-35247780-cecd-11e8-9c34-88b08fc39141.PNG)
